pkg/shared/findingkey/malware.go

```diff
@@ -19,22 +19,25 @@ import (
 	"fmt"
 
 	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/pkg/shared/utils"
 )
 
 type MalwareKey struct {
 	MalwareName string
 	MalwareType string
 	Path        string
+	RuleName    string
 }
 
 func (k MalwareKey) String() string {
-	return fmt.Sprintf("%s.%s.%s", k.MalwareName, k.MalwareType, k.Path)
+	return fmt.Sprintf("%s.%s.%s.%s", k.MalwareName, k.MalwareType, k.Path, k.RuleName)
 }
 
 func GenerateMalwareKey(info models.MalwareFindingInfo) MalwareKey {
 	return MalwareKey{
-		MalwareName: *info.MalwareName,
-		MalwareType: *info.MalwareType,
-		Path:        *info.Path,
+		MalwareName: utils.StringPointerValOrEmpty(info.MalwareName),
```

Comment:
Please use `utils.ValueOrZero` instead of `utils.StringPointerValOrEmpty`. Latter is obsoleted by the former.

```suggestion
		MalwareName: utils.ValueOrZero(info.MalwareName),
```